api/handlers.go

```diff
@@ -341,6 +341,29 @@ func getVolumes(c *context, w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
+	danglingOnly := false
+	refVolumes := make(map[string]bool)
+
+	if filters.Include("dangling") {
+		if filters.ExactMatch("dangling", "true") {
+			danglingOnly = true
```

Comment:
Note that the `dangling` filter also supports `dangling=false`, which returns all volumes that are _not_ dangling (that case could be handled by passing the filter to the engine, and collect all non-dangling volumes)

Reply:
Right, calling the engine directly was one simple option (see PR description), but api calls to all engines in the cluster can be taxing, hence we decided to build the list of used volumes from locally cached containers.

Reply:
Won't that easily lead to incorrect data (i.e. container may now use the volume, or vice-versa); or not a problem?

Reply:
good catch @thaJeztah! I will address the above